cosmos/precompile/bank/bank.go

```diff
@@ -254,16 +274,41 @@ func (c *Contract) Send(
 	if err != nil {
 		return false, err
 	}
+	fromAddr, err := cosmlib.StringFromEthAddress(c.addressCodec, fromAddress)
+	if err != nil {
+		return false, err
+	}
 	toAddr, err := cosmlib.StringFromEthAddress(c.addressCodec, toAddress)
 	if err != nil {
 		return false, err
 	}
 
-	_, err = c.msgServer.Send(ctx, &banktypes.MsgSend{
-		FromAddress: caller,
+	var msg sdk.Msg = &banktypes.MsgSend{
+		FromAddress: fromAddr,
 		ToAddress:   toAddr,
 		Amount:      amount,
-	})
+	}
+	if caller != fromAddr {
+		inner, err := cdctypes.NewAnyWithValue(msg)
+		if err != nil {
+			return false, err
+		}
+
+		msg = &authztypes.MsgExec{
+			Grantee: caller,
+			Msgs:    []*cdctypes.Any{inner},
+		}
+	}
+
+	handler := c.msgRouter.Handler(msg)
```

Comment:
@calbera fuck this might've worked

Reply:
let's revive authz too!

Reply:
yeah this makes a lot of sense. moving towards the direction of the precompileModule with this one. each precompile has a router to pass arbitrary messages